### PR TITLE
Fix issue 414 for TransferGroup. Add tests for Groups

### PR DESCRIPTION
**Pull request by bzon**
__Work in Progress__ - Add test for all groups.go api.

Fixes issue https://github.com/xanzy/go-gitlab/issues/414.

### CODE REVIEW

File: groups.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func (s *GroupsService) CreateGroup(opt *CreateGroupOptions, options ...OptionFu
 //
 // GitLab API docs:
 // https://docs.gitlab.com/ce/api/groups.html#transfer-project-to-group
-func (s *GroupsService) TransferGroup(gid interface{}, project int, options ...OptionFunc) (*Group, *Response, error) {
+func (s *GroupsService) TransferGroup(gid interface{}, project interface{}, options ...OptionFunc) (*Group, *Response, error) {
```
Comment (svanharmelen) [inline groups.go]: Nice! You should also add something like `project, err := parseID(pid)` to make sure the ID is parsed correctly 👍

Comment (svanharmelen) [reply groups.go]: And then maybe rename the argument from `project` to `pid` to be inline with other ode in this package. Thanks!

**Comment by bzon** [comment]
@svanharmelen This is ready for review. I also applied your suggested change. 👍

**Comment by svanharmelen** [comment]
Looks good @bzon! Your on 🔥

Thanks!

**Pull request merged**